### PR TITLE
Rename unqualified method calls

### LINKED ISSUE #11281: Rename symbol handler does not rename symbol usages
Applying an edit in GUI appears to only change the function's definition name but not its usages:
[Kazam_screencast_00014.webm](https://github.com/user-attachments/assets/53764c68-0702-4a58-8417-2448dfde5ec8)

Tracked it down to [RefactoringRenameJob](https://github.com/enso-org/enso/blob/develop/engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/RefactoringRenameJob.scala#L110) which only infers edits for the function name.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #11281

Changelog:
- update: `refactoring/renameSymbol` can rename unqualified method calls

### Important Notes

https://github.com/user-attachments/assets/e03c4ec7-7620-4ce4-8eab-86b95f308be2

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: engine/runtime-compiler/src/main/scala/org/enso/compiler/refactoring/IRUtils.scala [hunk 1]
```diff
@@ -64,25 +65,39 @@ trait IRUtils {
...
+              resolution.target match {
+                case BindingsMap.ResolvedModuleMethod(module, _)
+                    if module.getName == moduleName =>
+                  Some(function)
+                case _ =>
+                  None
+              }
+            case None =>
+              args.headOption match {
+                case Some(arg) if isSyntheticArgument(arg) =>
```
Comment (hubertp) [inline engine/runtime-compiler/src/main/scala/org/enso/compiler/refactoring/IRUtils.scala]: Do we not set any flag that the argument is synthetic? I kind of dislike comparing the code and `<...>`

Comment (4e6) [reply engine/runtime-compiler/src/main/scala/org/enso/compiler/refactoring/IRUtils.scala]: No, the argument doesn't even have such a field.

Comment (4e6) [reply engine/runtime-compiler/src/main/scala/org/enso/compiler/refactoring/IRUtils.scala]: Now it has https://github.com/enso-org/enso/pull/11556/commits/b077070e5ec7e338244dc615d160becf5b25e667

Comment (hubertp) [reply engine/runtime-compiler/src/main/scala/org/enso/compiler/refactoring/IRUtils.scala]: It's one of those "be careful what you wish for" changes

**Review by hubertp — APPROVED** [review]

**Pull request merged**